### ISSUE TITLE
Using SiriProxy over 3G

### ISSUE THREAD

**Issue by ghost** [plamoni/SiriProxy#282]
So now that the 4S is jailbroken, theoretically we should be able to access SiriProxy through our 3G by adding an entry to the /etc/hosts file.  I did that and I get nothing.  What am I doing wrong/not doing at all??

My IP address has been replaced with stars for purposes of this post.
## 
# Host Database
# 
# localhost is used to configure the loopback interface
# when the system is booting.  Do not change this entry.
## 

127.0.0.1   localhost
255.255.255.255 broadcasthost
::1             localhost 
fe80::1%lo0 localhost
_._._._ guzzoni.apple.com

*edit: my friend was able to succesfully accomplish this using SiriSpire on the 3Gs

**Comment by Winter-Guerra** [comment]
I'm working on getting a Cydia tool working for redirecting Siri to a proxy. Should be done later today or tomorrow. 

In the meantime, I don't see any stars anywhere. Did you put your ip address in the host file in the form:
ip.ip.ip.ip guzzoni.apple.com

**Comment by ghost** [comment]
Sweet! Let me know when it's ready man, it'd be much appreciated.  Yes, I did, the stars were removed from the comment? It's weird.

**Comment by ntetreau** [comment]
I was able to get this woking on the 4S by adding this line to the /etc/hosts fiel:

myserver.dyndns.org guzzoni.apple.com

then using ifile, go into /var/mobile/Library/Preferences -> com.apple.assistant.plist (edit with Property List Viewer)
Accounts > [lots of numbers] > Hostname
and change the server to https://myserver.dyndns.org

**Comment by ghost** [comment]
It seems to connect now but I get: 
Create server for iPhone connection
start conn #, @zip_stream=#Zlib::Deflate:0xa37c7b0, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins laoded: [#>]

And it just hangs. Siri tells me that it can't do anything and to check back later.

**Comment by Winter-Guerra** [comment]
Could you post a copy of your new .plist? (with your keys blacked out) I think you might be missing a field or something. Did you remember to use https instead of http?

**Comment by milosGR** [comment]
Ok so I just setup my server and I'm forwarding 443 and 53 from my router to virtualbox Ubuntu SiriProxy that I just setup... also I edited /etc/hosts on my iphone4s and the plist file mentioned above with my Public IP address of my router but Siri says "really sorry about this.. somethings wrong bla bla" and on my SiriProxy I see the below:

Create server for iPhone connection
start conn #<SiriProxy::Connection::Iphone:0x8a38984 @signature=81, @processed_headers=false, @output_buffer="", @input_buffer="", @unzipped_input="", @unzipped_output="", @unzip_stream=#<Zlib::Inflate:0x8a3881c>, @zip_stream=#Zlib::Deflate:0x8a387f4, @consumed_ace=false, @name="iPhone", @ssled=false>
[Info - Plugin Manager] Plugins laoded: [#<SiriProxy::Plugin::Example:0x8a37890 @manager=#<SiriProxy::PluginManager:0x8a383e4 @plugins=[...]>>]

**Comment by milosGR** [comment]
Update.. I got it working!  Will post how I resolved it shortly..

**Comment by Winter-Guerra** [comment]
Awesomeness! Please do tell us how you did it!

In other news, I got my Cydia tweak working and am now working hard now at making a decent icon for it and uploading it to Cydia's bigboss repo as fast as I can. In the meantime check out these development screenshots of SiriSwitch here: http://www.flickr.com/photos/60598968@N04/sets/72157629009104211/

Just flick the switch to go from SiriProxy to Guzzoni and back again! On 3G, EDGE or WIFI!

**Comment by ntetreau** [comment]
@xtremd SiriSwitch looks very interesting, could you elaborate on what is does in terms of the plist host and the /etc/hosts file when it is switched on and off?

**Comment by milosGR** [comment]
xtremd,
To fix it I had to put both names in the cert (dyndns & guzzoni) per the instructions in the script located here:  https://gist.github.com/1552448

**Comment by Winter-Guerra** [comment]
This thread is probably dead by now, but I've compiled, tested, and submitted the tweak to Cydia's bigboss repo. They say it should take less than 24 hours for it to be synced into their repo and become available to iPhone4S users. SiriSwitch is on it's way! Hopefully you guys should find it useful for quickly switching between the proxies and Apple's Guzzoni servers.

@ntetreau If you're interested, the code for the tweak can be found here: https://github.com/xtremd/SiriSwitch
Basically, the tweak utilizes MobileSubstrate to catch and then override Siri's internal system call for the Siri hostname. It then redirects Siri to the domain of your choice based on the settings you set in the settings pane. This way, the plist never gets touched and Siri alone is redirected through the proxy server.

**Comment by shanagarwal** [comment]
Hi,

I installed SiriProxy in ubuntu 11.10, generated ca.crt with dual cn and sent it to iPhone 4 & 4s.
I'm able to connect to siriproxy over 3g through my 4 but not through my 4s.

Any suggestions will be appreciated greatly.
Thanks in advance.

**Comment by TobyTech** [comment]
Where do I direct my dyndns server to? External? Internal?

**Comment by ntetreau** [comment]
On the 4 and 4s, I have the same /etc/hosts 
proxyserver.dyndns.org guzzoni.apple.com

And in ifile /var/mobile/library/preferences assistant.plist
Hostname pointing to proxyserver.dyndns.org

That combination seems to work perfectly for me.

**Comment by ntetreau** [comment]
Host name in the plist is actually:
Https://proxyserver.dyndns.org

**Comment by TobyTech** [comment]
What do I put for the IP in dyndns? External? Internal?

**Comment by ntetreau** [comment]
External, then port forward 443 from your router to the internal ip of your siriproxy

**Comment by TobyTech** [comment]
Dns wont let me add a port to the hostname.

Ip.ip.ip.ip:443.    Is invalid. "must enter a valid IP"

**Comment by ntetreau** [comment]
Dont add the port to the DNs, just put your external ip. Then on your router, forward port 443 to your internal ip

**Comment by TobyTech** [comment]
Alright, I give up, I cannot find how I forward port 443 to my internal IP.

**Comment by TobyTech** [comment]
Alright, is this the correct thingy? http://i.imgur.com/H7Lwq.png

**Comment by Begall** [comment]
@xtremd Has bigboss rejected your submission? I've had a look and I can't see it on there yet. Great work btw!

**Comment by milosGR** [comment]
@xtremd I can't find it on bigboss either...  what happened?

**Comment by Winter-Guerra** [comment]
@milosGR @Begall Optimo (the developer running the Bigboss repo) did not reject my submission, but instead encouraged me to contact Chpwn and look into merging my changes into Chpwn's Spire tweak to avoid repetitive packages cluttering up their server. He thought it would be nice if Spire was /the/ one-stop Siri hacking utility and, I must say, I agree. That said, I have been meaning to construct and submit a pull request to Chpwn, but have not had time to go through with my promise due to an overwhelmingly busy week of school. In the meantime, I have loaded SiriSwitch for your hacking pleasure onto my own private repo here: 
http://xtremd.no-ip.org/cydia/repo/

Just go to Cydia->Manage->Edit->Add repo and add the URL to your list of repos. Then search for SiriSwitch in the Cydia search box and install.

Please, tell me what you think and thanks for all the interest!

**Comment by TobyTech** [comment]
@ntetreau  Alright. I did the https://gist.github.com/1552448  thing, and SiriProxy is up and running. I also have the certs installed on my iPhone set up correctly. I have a Verizon MI424 router, and would like to know what I have to do to be able to have my iPhone (iPad later on) connect to the server over 3G and other Wifi networks besides local. What do I have to do now?

**Comment by ghost** [comment]
Has anybody been able to get this going?  Even with the switch and certs I still get a connection that just closes quickly.

**Comment by worldjoe** [comment]
xtremd: would be helpful to have SiriSwitch in a repo unrelated to Spire. Some of us have Siri without the need for Spire, but still want to proxy our Siri calls so we can use cool plugins. I tried pulling down your source, your makefile references lots of hardcoded paths for other makefiles. Not sure what they're trying to pull in.

**Comment by elvisimprsntr** [comment]
I use open source router firmware from www.dd-wrt.com which supports VPN server and an transparent DNS redirect.  To use SiriProxy all I have to do is turn on VPN and route all traffic thru the VPN.    It does not require a JB and eliminates the need to manually set the DNS on the iPhone.  You will need to set up DYNDNS in the router with www.no-ip.com or similar to resolve your ISP assigned IP address to a hostname.
